src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
Agree, yes. Although "Can't repurpose preexisting or third party attributes." is just a con, not both a pro and a con.
The more specific error checking is a potential improvement (not that it is already doing it), and might not be that useful anyway, since it's probably very trivial to see that you didn't configure any storage provider once it fails to resolve the storage object. But yes, it's a very minor con.